### ISSUE TITLE
LVM Backend is broken on 0.26

### ISSUE THREAD

**Issue by alitvak69** [lxc/lxd#1470]
Sorry everyone,

Didn't mean to double post but thought the issue need to be separated.

I just tested 0.26 and no such luck. LVM is still broken when creating images with lxd

lxc init centos7-amd64 webtest1-chi -p default
error: Error mounting snapshot LV: no such file or directory

```
lxc init --debug --verbose centos7-amd64 webtest2-chi -p default
Creating webtest2-chi DBUG[01-06|02:41:55] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"api_compat":1,"auth":"trusted","config":{"storage.lvm_thinpool_name":"LXDPool","storage.lvm_vg_name":"lxc"},"environment":{"addresses":[],"architectures":[2,1],"driver":"lxc","driver_version":"1.1.5","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.2.0-23-generic","server":"lxd","server_pid":1508,"server_version":"0.26","storage":"lvm","storage_version":"2.02.122(2) (2015-06-20) / 1.02.99 (2015-06-20) / 4.33.0"}},"operation":""}
 
DBUG[01-06|02:41:55] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"target":"1582f9bc9f9558f826cad94dc8968004c8c014c4bfcffcadfbd1e4e96145404e","description":"centos7-amd64"},"operation":""}
 
DBUG[01-06|02:41:55] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"aliases":[{"target":"centos7-amd64","description":"centos7-amd64"}],"architecture":2,"fingerprint":"1582f9bc9f9558f826cad94dc8968004c8c014c4bfcffcadfbd1e4e96145404e","filename":"centos-7-amd64-default-20160105_02:16.tar.xz","properties":{"description":"Centos 7 (amd64)"},"public":0,"size":61906508,"created_at":1451964217,"expires_at":0,"uploaded_at":1452047788},"operation":""}
 
DBUG[01-06|02:41:55] Posting {"name":"webtest2-chi","profiles":["default"],"source":{"fingerprint":"1582f9bc9f9558f826cad94dc8968004c8c014c4bfcffcadfbd1e4e96145404e","type":"image"}}
 to http://unix.socket/1.0/containers 
DBUG[01-06|02:41:55] Raw response: {"type":"async","status":"OK","status_code":100,"metadata":{"id":"37cd6b43-67b4-4ef6-8a1f-49c61ec9b2db","class":"task","created_at":"2016-01-06T02:41:55.866914316Z","updated_at":"2016-01-06T02:41:55.866914316Z","status":"Running","status_code":103,"resources":{"containers":["/1.0/containers/webtest2-chi"]},"metadata":null,"may_cancel":false,"err":""},"operation":"/1.0/operations/37cd6b43-67b4-4ef6-8a1f-49c61ec9b2db"}
 
DBUG[01-06|02:41:55] 1.0/operations/37cd6b43-67b4-4ef6-8a1f-49c61ec9b2db/wait 
DBUG[01-06|02:42:06] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"id":"37cd6b43-67b4-4ef6-8a1f-49c61ec9b2db","class":"task","created_at":"2016-01-06T02:41:55.866914316Z","updated_at":"2016-01-06T02:41:55.866914316Z","status":"Failure","status_code":400,"resources":{"containers":["/1.0/containers/webtest2-chi"]},"metadata":null,"may_cancel":false,"err":"Error mounting snapshot LV: no such file or directory"},"operation":""}
```
```
lxc config show
config:
  storage.lvm_thinpool_name: LXDPool
  storage.lvm_vg_name: lxc
```
```
lxc profile show default
name: default
config:
  raw.lxc: |
    lxc.start.auto = 1
    lxc.arch = amd64
    lxc.cgroup.cpu.shares = 1
  security.privileged: "true"
devices:
  eth0:
    nictype: bridged
    parent: br0
    type: nic
```

**Comment by alitvak69** [comment]
Posting lvm info
```
lvs
  LV                                                               VG   Attr       LSize   Pool    Origin Data%  Meta%  Move Log Cpy%Sync Convert
  1582f9bc9f9558f826cad94dc8968004c8c014c4bfcffcadfbd1e4e96145404e lxc  Vwi-a-tz--  10.00g LXDPool        6.94                                   
  LXDPool                                                          lxc  twi-aotz-- 152.84g                0.45   0.10                            
  xbroker8-chi                                                     lxc  -wi-a-----  80.00g                                                       
  lxcbase                                                          vg00 -wi-ao---- 235.84g                                                       
  root                                                             vg00 -wi-ao----  27.94g     
```
```
vgs
  VG   #PV #LV #SN Attr   VSize   VFree
  lxc    1   3   0 wz--n- 235.84g 1.00g
  vg00   1   2   0 wz--n- 263.78g    0 
```

**Comment by tych0** [comment]
Looks like this is also mentioned in #1467. I'll take a look right now, because I bet it was some of the new snapshot transfer code I added that is breaking/broke this.

**Comment by tych0** [comment]
Oh, whoops, looks like I was misunderstanding things. This seems to be a totally different issue :)

**Comment by tych0** [comment]
Can you post the LXD logs when this happens? Also, can you show the output of `tree /dev/lxc`?

**Comment by alitvak69** [comment]
```
/dev/lxc
|-- d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 -> ../dm-7
`-- xbroker8-chi -> ../dm-2
```
less /var/log/lxd/lxd.log 
```
t=2016-01-03T17:32:38+0000 lvl=info msg="LXD is starting" path=/var/lib/lxd
t=2016-01-03T17:32:38+0000 lvl=info msg="Default uid/gid map:"
t=2016-01-03T17:32:38+0000 lvl=info msg=" - u 0 165536 65536"
t=2016-01-03T17:32:38+0000 lvl=info msg=" - g 0 165536 65536"
t=2016-01-03T17:32:38+0000 lvl=info msg=Init driver=storage/dir
t=2016-01-03T17:32:38+0000 lvl=info msg="Looking for existing certificates" key=/var/lib/lxd/server.key cert=/var/lib/lxd/server.crt
t=2016-01-03T17:32:49+0000 lvl=info msg="LXD is socket activated"
t=2016-01-03T17:32:49+0000 lvl=info msg="REST API daemon:"
t=2016-01-03T17:32:49+0000 lvl=info msg=" - binding socket" socket=/var/lib/lxd/unix.socket
t=2016-01-03T18:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-03T19:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-03T20:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-03T21:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-03T22:17:02+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-03T23:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-04T00:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-04T01:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-04T01:58:51+0000 lvl=info msg="Received 'power failure signal', shutting down containers."
t=2016-01-04T01:58:51+0000 lvl=info msg="Stopping REST API handler:"
t=2016-01-04T01:58:51+0000 lvl=info msg=" - skipping socket-activated socket" socket=/var/lib/lxd/unix.socket
t=2016-01-04T01:58:56+0000 lvl=info msg="LXD is starting" path=/var/lib/lxd
t=2016-01-04T01:58:56+0000 lvl=warn msg="CGroup memory swap accounting is disabled, swap limits will be ignored."
t=2016-01-04T01:58:56+0000 lvl=info msg="Default uid/gid map:"
t=2016-01-04T01:58:56+0000 lvl=info msg=" - u 0 165536 65536"
t=2016-01-04T01:58:56+0000 lvl=info msg=" - g 0 165536 65536"
t=2016-01-04T01:58:56+0000 lvl=info msg=Init driver=storage/dir
t=2016-01-04T01:58:56+0000 lvl=info msg="Looking for existing certificates" cert=/var/lib/lxd/server.crt key=/var/lib/lxd/server.key
t=2016-01-04T01:58:56+0000 lvl=info msg="LXD is socket activated"
t=2016-01-04T01:58:56+0000 lvl=info msg="REST API daemon:"
t=2016-01-04T01:58:56+0000 lvl=info msg=" - binding socket" socket=/var/lib/lxd/unix.socket
t=2016-01-04T01:58:56+0000 lvl=info msg=handling method=GET url=/1.0 ip=@
t=2016-01-04T01:58:56+0000 lvl=info msg=handling method=GET url=/1.0 ip=@
t=2016-01-04T02:17:01+0000 lvl=info msg=handling method=GET url="/1.0/images?recursion=1" ip=@
t=2016-01-04T02:51:27+0000 lvl=info msg="Received 'power failure signal', shutting down containers."
t=2016-01-04T02:51:27+0000 lvl=info msg="Stopping REST API handler:"
t=2016-01-04T02:51:27+0000 lvl=info msg=" - skipping socket-activated socket" socket=/var/lib/lxd/unix.socket
t=2016-01-04T03:58:43+0000 lvl=info msg="LXD is starting" path=/var/lib/lxd
t=2016-01-04T03:58:43+0000 lvl=warn msg="CGroup memory swap accounting is disabled, swap limits will be ignored."
...skipping...
\t\"metadata\": {\n\t\t\t\"target\": \"d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5\",\n\t\t\t\"description\": \"centos7-amd64\"\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
t=2016-01-06T21:08:57+0000 lvl=info msg=handling method=GET url=/1.0/images/d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 ip=@
t=2016-01-06T21:08:57+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"aliases\": [\n\t\t\t\t{\n\t\t\t\t\t\"target\": \"centos7-amd64\",\n\t\t\t\t\t\"description\": \"centos7-amd64\"\n\t\t\t\t}\n\t\t\t],\n\t\t\t\"architecture\": 2,\n\t\t\t\"fingerprint\": \"d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5\",\n\t\t\t\"filename\": \"centos-7-amd64-default-20160106_02:16.tar.xz\",\n\t\t\t\"properties\": {\n\t\t\t\t\"description\": \"Centos 7 (amd64)\"\n\t\t\t},\n\t\t\t\"public\": 0,\n\t\t\t\"size\": 61926024,\n\t\t\t\"created_at\": 1452050577,\n\t\t\t\"expires_at\": 0,\n\t\t\t\"uploaded_at\": 1452114511\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
t=2016-01-06T21:08:57+0000 lvl=info msg=handling method=POST url=/1.0/containers ip=@
t=2016-01-06T21:08:57+0000 lvl=dbug msg="\n\t{\n\t\t\"name\": \"webtest2-chi\",\n\t\t\"profiles\": [\n\t\t\t\"default\"\n\t\t],\n\t\t\"source\": {\n\t\t\t\"fingerprint\": \"d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5\",\n\t\t\t\"type\": \"image\"\n\t\t}\n\t}"
t=2016-01-06T21:08:57+0000 lvl=dbug msg="Responding to container create"
t=2016-01-06T21:08:57+0000 lvl=dbug msg="New task operation: 65c09dbe-941b-43f7-b392-39d91ce296b2"
t=2016-01-06T21:08:57+0000 lvl=dbug msg="Started task operation: 65c09dbe-941b-43f7-b392-39d91ce296b2"
t=2016-01-06T21:08:57+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"async\",\n\t\t\"status\": \"OK\",\n\t\t\"status_code\": 100,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"65c09dbe-941b-43f7-b392-39d91ce296b2\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-06T21:08:57.839683294Z\",\n\t\t\t\"updated_at\": \"2016-01-06T21:08:57.839683294Z\",\n\t\t\t\"status\": \"Running\",\n\t\t\t\"status_code\": 103,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest2-chi\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"\"\n\t\t},\n\t\t\"operation\": \"/1.0/operations/65c09dbe-941b-43f7-b392-39d91ce296b2\"\n\t}"
t=2016-01-06T21:08:57+0000 lvl=info msg=handling method=GET url=/1.0/operations/65c09dbe-941b-43f7-b392-39d91ce296b2/wait ip=@
t=2016-01-06T21:08:57+0000 lvl=dbug msg=ContainerCreateFromImage driver=storage/lvm imageFingerprint=d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 name=webtest2-chi isPrivileged=true
t=2016-01-06T21:09:03+0000 lvl=dbug msg=containerDeleteSnapshots container=webtest2-chi
t=2016-01-06T21:09:03+0000 lvl=dbug msg=ContainerDelete driver=storage/lvm container=webtest2-chi
t=2016-01-06T21:09:08+0000 lvl=dbug msg="Could not remove LV" driver=storage/lvm lvname=webtest2--chi output="  Failed to find logical volume \"lxc/webtest2--chi\"\n"
t=2016-01-06T21:09:08+0000 lvl=dbug msg="Failure for task operation: 65c09dbe-941b-43f7-b392-39d91ce296b2: Error mounting snapshot LV: no such file or directory"
t=2016-01-06T21:09:08+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"65c09dbe-941b-43f7-b392-39d91ce296b2\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-06T21:08:57.839683294Z\",\n\t\t\t\"updated_at\": \"2016-01-06T21:08:57.839683294Z\",\n\t\t\t\"status\": \"Failure\",\n\t\t\t\"status_code\": 400,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest2-chi\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"Error mounting snapshot LV: no such file or directory\"\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
```
less /var/log/lxd/webtest2-chi/lxc.log
```
 lxc 1452114537.859 WARN     lxc_cgmanager - cgmanager.c:cgm_get:994 - do_cgm_get exited with error
```

**Comment by alitvak69** [comment]
It attempts to delete snapshots on creation of new containers and fails after not being able to find them.

**Comment by alitvak69** [comment]
I edited the previous message since my terminal skipped on bunch of log lines.

**Comment by tych0** [comment]
I'm guessing that the create fails, and then it deletes it because of that; i.e. it's not trying to delete things when it's creating the container, but only after it's failed to do that.

**Comment by tych0** [comment]
I'm not really sure what's going on here. It looks like the `lvcreate` command is succeeding, but not actually creating the device somehow. If I sent a patch with a bunch of debug statements, could you build that and test with it so we can get a better picture?

**Comment by alitvak69** [comment]
I can do it.  One thing that is interesting, lvcreate i successful with
creating os image but not with creating container image.

**Comment by mikemccracken** [comment]
@alitvak69 the initial error message I see in your log says this: 
```t=2016-01-06T21:07:17+0000 lvl=dbug msg="Could not create LV" driver=storage/lvm lvname=d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 output="  Pool LXDPool not found in Volume group lxc.\n"
```

However your comment with the LVM info shows that there is a thin pool named LXDPool in vg lxc.
When did you run 'lvs'? Is it possible that the pool was created after that initial failure?

It seems like there might be a bug here, but if the pool did exist before you tried to create the image, then we need to figure out why lvcreate seems confused here.

**Comment by alitvak69** [comment]
I actually have to cleanup everything manually.  I remove all images and
container files, then container and images related entries from lxd.db,
and finally I remove image volumes and thin pool.  I only leave config
pointing to vg lxc since it is non-default  group name for lxd.

When I retry I recreate local centos7 image and then init a container.  At
some point thin pool LXDPool is automatically created for me.

**Comment by alitvak69** [comment]
In addition I wanted to say that I posted the whole log from today and perhaps at some point I forgot to unset config for LXDPool but removed it with lvremove.  Whence lxd assumed it was already there.  

I cleaned this up retried creating local image first and then creating a fresh container based on local image.  It would probably make sense if you concentrated on last lines of the log.

**Comment by alitvak69** [comment]
Here is my last lvs and vgs after I cleaned it all up.

```
vgs
  VG   #PV #LV #SN Attr   VSize   VFree  
  lxc    1   1   0 wz--n- 235.84g 155.84g
  vg00   1   2   0 wz--n- 263.78g 

lvs
  LV           VG   Attr       LSize   Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  xbroker8-chi lxc  -wi-a-----  80.00g                                                    
  lxcbase      vg00 -wi-ao---- 235.84g                                                    
  root         vg00 -wi-ao----  27.94g 
```

**Comment by alitvak69** [comment]
Here is lvs before lxc init
```
lvs
  LV                                                               VG   Attr       LSize   Pool    Origin Data%  Meta%  Move Log Cpy%Sync Convert
  LXDPool                                                          lxc  twi-aotz-- 152.84g                0.45   0.10                            
  d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 lxc  Vwi-a-tz--  10.00g LXDPool        6.94                                   
  xbroker8-chi                                                     lxc  -wi-a-----  80.00g                                                       
  lxcbase                                                          vg00 -wi-ao---- 235.84g                                                       
  root                                                             vg00 -wi-ao----  27.94g     
```

**Comment by alitvak69** [comment]
Here is lvs after failed init
```
lvs
  LV                                                               VG   Attr       LSize   Pool    Origin Data%  Meta%  Move Log Cpy%Sync Convert
  LXDPool                                                          lxc  twi-aotz-- 152.84g                0.45   0.10                            
  d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 lxc  Vwi-a-tz--  10.00g LXDPool        6.94                                   
  xbroker8-chi                                                     lxc  -wi-a-----  80.00g                                                       
  lxcbase                                                          vg00 -wi-ao---- 235.84g                                                       
  root                                                             vg00 -wi-ao----  27.94g
```

**Comment by mikemccracken** [comment]
Off the top of my head, does it make a difference if you try with a different VG, one that has no pre-existing LVs in it before LXD tries to create LXDPool? my hunch is that xbroker8-chi might be interfering with the creation of LXDPool, but I haven't looked at that code in a bit so I don't know if it's changed.

If possible could you paste just the part of the log from the failed init, or just what the timestamps are on those runs of vgs/lvs/etc?

I am leaving the office now but can look into this later and knowing which log lines are relevant would be very helpful.

**Comment by alitvak69** [comment]
I cannot do it as my space is limited.  I guess creating another nested pv would work.  It really would be good to know the reason for such limitation (doesn't make sense to me), especially if os image is added and thinpool is created with no problems.

As far as the log goes, here you go, another init try.
```
cat lxd.log 

t=2016-01-06T23:52:14+0000 lvl=info msg=handling method=GET url=/1.0 ip=@
t=2016-01-06T23:52:14+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"api_compat\": 1,\n\t\t\t\"auth\": \"trusted\",\n\t\t\t\"config\": {\n\t\t\t\t\"storage.lvm_thinpool_name\": \"LXDPool\",\n\t\t\t\t\"storage.lvm_vg_name\": \"lxc\"\n\t\t\t},\n\t\t\t\"environment\": {\n\t\t\t\t\"addresses\": [],\n\t\t\t\t\"architectures\": [\n\t\t\t\t\t2,\n\t\t\t\t\t1\n\t\t\t\t],\n\t\t\t\t\"driver\": \"lxc\",\n\t\t\t\t\"driver_version\": \"1.1.5\",\n\t\t\t\t\"kernel\": \"Linux\",\n\t\t\t\t\"kernel_architecture\": \"x86_64\",\n\t\t\t\t\"kernel_version\": \"4.2.0-23-generic\",\n\t\t\t\t\"server\": \"lxd\",\n\t\t\t\t\"server_pid\": 2981,\n\t\t\t\t\"server_version\": \"0.26\",\n\t\t\t\t\"storage\": \"lvm\",\n\t\t\t\t\"storage_version\": \"2.02.122(2) (2015-06-20) / 1.02.99 (2015-06-20) / 4.33.0\"\n\t\t\t}\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
t=2016-01-06T23:52:14+0000 lvl=info msg=handling url=/1.0/images/aliases/centos7-amd64 ip=@ method=GET
t=2016-01-06T23:52:14+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"target\": \"d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5\",\n\t\t\t\"description\": \"centos7-amd64\"\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
t=2016-01-06T23:52:14+0000 lvl=info msg=handling method=GET url=/1.0/images/d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 ip=@
t=2016-01-06T23:52:14+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"aliases\": [\n\t\t\t\t{\n\t\t\t\t\t\"target\": \"centos7-amd64\",\n\t\t\t\t\t\"description\": \"centos7-amd64\"\n\t\t\t\t}\n\t\t\t],\n\t\t\t\"architecture\": 2,\n\t\t\t\"fingerprint\": \"d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5\",\n\t\t\t\"filename\": \"centos-7-amd64-default-20160106_02:16.tar.xz\",\n\t\t\t\"properties\": {\n\t\t\t\t\"description\": \"Centos 7 (amd64)\"\n\t\t\t},\n\t\t\t\"public\": 0,\n\t\t\t\"size\": 61926024,\n\t\t\t\"created_at\": 1452050577,\n\t\t\t\"expires_at\": 0,\n\t\t\t\"uploaded_at\": 1452124320\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
t=2016-01-06T23:52:14+0000 lvl=info msg=handling method=POST url=/1.0/containers ip=@
t=2016-01-06T23:52:14+0000 lvl=dbug msg="\n\t{\n\t\t\"name\": \"webtest21\",\n\t\t\"profiles\": [\n\t\t\t\"default\"\n\t\t],\n\t\t\"source\": {\n\t\t\t\"fingerprint\": \"d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5\",\n\t\t\t\"type\": \"image\"\n\t\t}\n\t}"
t=2016-01-06T23:52:14+0000 lvl=dbug msg="Responding to container create"
t=2016-01-06T23:52:14+0000 lvl=dbug msg="New task operation: 9b82b9ae-4962-4a8e-a770-cf595baeea3c"
t=2016-01-06T23:52:14+0000 lvl=dbug msg="Started task operation: 9b82b9ae-4962-4a8e-a770-cf595baeea3c"
t=2016-01-06T23:52:14+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"async\",\n\t\t\"status\": \"OK\",\n\t\t\"status_code\": 100,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"9b82b9ae-4962-4a8e-a770-cf595baeea3c\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-06T23:52:14.596529887Z\",\n\t\t\t\"updated_at\": \"2016-01-06T23:52:14.596529887Z\",\n\t\t\t\"status\": \"Running\",\n\t\t\t\"status_code\": 103,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest21\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"\"\n\t\t},\n\t\t\"operation\": \"/1.0/operations/9b82b9ae-4962-4a8e-a770-cf595baeea3c\"\n\t}"
t=2016-01-06T23:52:14+0000 lvl=info msg=handling method=GET url=/1.0/operations/9b82b9ae-4962-4a8e-a770-cf595baeea3c/wait ip=@
t=2016-01-06T23:52:14+0000 lvl=dbug msg=ContainerCreateFromImage driver=storage/lvm imageFingerprint=d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 name=webtest21 isPrivileged=true
t=2016-01-06T23:52:19+0000 lvl=dbug msg=containerDeleteSnapshots container=webtest21
t=2016-01-06T23:52:19+0000 lvl=dbug msg=ContainerDelete driver=storage/lvm container=webtest21
t=2016-01-06T23:52:24+0000 lvl=dbug msg="Could not remove LV" driver=storage/lvm output="  Failed to find logical volume \"lxc/webtest21\"\n" lvname=webtest21
t=2016-01-06T23:52:24+0000 lvl=dbug msg="Failure for task operation: 9b82b9ae-4962-4a8e-a770-cf595baeea3c: Error mounting snapshot LV: no such file or directory"
t=2016-01-06T23:52:24+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"9b82b9ae-4962-4a8e-a770-cf595baeea3c\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-06T23:52:14.596529887Z\",\n\t\t\t\"updated_at\": \"2016-01-06T23:52:14.596529887Z\",\n\t\t\t\"status\": \"Failure\",\n\t\t\t\"status_code\": 400,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest21\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"Error mounting snapshot LV: no such file or directory\"\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
```

**Comment by alitvak69** [comment]
I have tried creating a nested vg lxd
```
lvs
  LV                                                               VG   Attr       LSize   Pool    Origin Data%  Meta%  Move Log Cpy%Sync Convert
  lxdbase                                                          lxc  -wi-ao---- 120.00g                                                       
  xbroker8-chi                                                     lxc  -wi-ao----  80.00g                                                       
  LXDPool                                                          lxd  twi-aotz-- 117.00g                0.59   0.09                            
  d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 lxd  Vwi-a-tz--  10.00g LXDPool        6.94                                   
  lxcbase                                                          vg00 -wi-ao---- 235.84g                                                       
  root                                                             vg00 -wi-ao----  27.94g    
```
Here is the log
```
t=2016-01-07T00:10:26+0000 lvl=dbug msg="Responding to container create"
t=2016-01-07T00:10:26+0000 lvl=dbug msg="New task operation: 86022414-a678-4985-a74a-4ae29192e3ad"
t=2016-01-07T00:10:26+0000 lvl=dbug msg="Started task operation: 86022414-a678-4985-a74a-4ae29192e3ad"
t=2016-01-07T00:10:26+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"async\",\n\t\t\"status\": \"OK\",\n\t\t\"status_code\": 100,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"86022414-a678-4985-a74a-4ae29192e3ad\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-07T00:10:26.036575312Z\",\n\t\t\t\"updated_at\": \"2016-01-07T00:10:26.036575312Z\",\n\t\t\t\"status\": \"Running\",\n\t\t\t\"status_code\": 103,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest21\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"\"\n\t\t},\n\t\t\"operation\": \"/1.0/operations/86022414-a678-4985-a74a-4ae29192e3ad\"\n\t}"
t=2016-01-07T00:10:26+0000 lvl=info msg=handling method=GET url=/1.0/operations/86022414-a678-4985-a74a-4ae29192e3ad/wait ip=@
t=2016-01-07T00:10:26+0000 lvl=dbug msg=ContainerCreateFromImage driver=storage/lvm imageFingerprint=d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5 name=webtest21 isPrivileged=true
t=2016-01-07T00:10:31+0000 lvl=dbug msg=containerDeleteSnapshots container=webtest21
t=2016-01-07T00:10:31+0000 lvl=dbug msg=ContainerDelete driver=storage/lvm container=webtest21
t=2016-01-07T00:10:36+0000 lvl=dbug msg="Could not remove LV" driver=storage/lvm lvname=webtest21 output="  Failed to find logical volume \"lxd/webtest21\"\n"
t=2016-01-07T00:10:36+0000 lvl=dbug msg="Failure for task operation: 86022414-a678-4985-a74a-4ae29192e3ad: Error mounting snapshot LV: no such file or directory"
t=2016-01-07T00:10:36+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"86022414-a678-4985-a74a-4ae29192e3ad\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-07T00:10:26.036575312Z\",\n\t\t\t\"updated_at\": \"2016-01-07T00:10:26.036575312Z\",\n\t\t\t\"status\": \"Failure\",\n\t\t\t\"status_code\": 400,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest21\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"Error mounting snapshot LV: no such file or directory\"\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
``

**Comment by mikemccracken** [comment]
I think I know what's going on with LXD here - it appears to be ignoring an error from LVM while creating the LV for the container... If you want to find out more about the error it's ignoring, you can try the same operation manually, with more verbose logging - e.g.: `sudo lvcreate -aay -v -n webtest21 -s /dev/lxc/d755ac26ef626574ffcf0213df1772d4739da3f3ff002f7fd3019288454117c5`

(Note, if that's not verbose enough for you, add `v`s to the `-v`

**Comment by alitvak69** [comment]
So is it a lxd issue or lvm?

**Comment by mikemccracken** [comment]
Both. LVM is throwing an error (I don't know what it is) and LXD is not
reporting that error, and continuing erroneously until something else fails.

**Comment by mickydelfavero** [comment]
I've the same issue (https://lists.linuxcontainers.org/pipermail/lxc-users/2015-December/010700.html sorry for not opening an issue that time), today I investigate and discover that starting from 0.25 something change in lv creation
On 0.24 `createSnapshotLV()` is defined like:

    func (s *storageLvm) createSnapshotLV(lvname string, origlvname string, readonly bool) (string, error) {
            output, err := exec.Command(
                    "lvcreate",
                    "-kn",
                    "-n", lvname,
                    "-s", fmt.Sprintf("/dev/%s/%s", s.vgName, origlvname)).CombinedOutput()
            if err != nil {

starting from 0.25 it's defined as follow:

    func (s *storageLvm) createSnapshotLV(lvname string, origlvname string, readonly bool) (string, error) {
            output, err := s.tryExec(
                    "lvcreate",
                    "-aay",
                    "-n", lvname,
                    "-s", fmt.Sprintf("/dev/%s/%s", s.vgName, origlvname))
            if err != nil {

`-aay` means autoactivation of LV, `man lvcreate` says that:

>        -a, --activate {y|ay|n|ey|en|ly|ln}
>              Controls the availability of the Logical Volumes for
>              immediate use after the command finishes running.  By
>              default, new Logical Volumes are activated (-ay).  If it
>              is possible technically, -an will leave the new Logical
>              Volume inactive. But for example, snapshots can only be
>              created in the active state so -an cannot be used with
>              --snapshot.  Normally the --zero n argument has to be
>              supplied too because zeroing (the default behaviour)
>              also requires activation.  If autoactivation option is
>              used (-aay), the logical volume is activated only if it
>              matches an item in the
>              activation/auto_activation_volume_list set in
>              lvm.conf(5).  For autoactivated logical volumes, --zero
>              n and --wipesignatures n is always assumed and it can't
>              be overridden. If the clustered locking is enabled, -aey
>              will activate exclusively on one node and -a{a|l}y will
>              activate only on the local node.

and `man 5 lvm.conf` explains that:

>              **volume_list** — This acts as a filter through which all
>              requests to activate a logical volume on this machine
>              are passed.  A logical volume is only activated if it
>              matches an item in the list.  Tags must be preceded by @
>              and are checked against all tags defined in the logical
>              volume and volume group meta‐ data for a match.  @* is
>              short-hand to check every tag set on the host machine
>              (see tags above).  Logi‐ cal volume and volume groups
>              can also be included in the list by name e.g. vg00,
>              vg00/lvol1.  If this setting is not present but at least
>              one host tag is defined then a default single-entry list
>              containing @* is assumed.
>
>              **auto_activation_volume_list** — This acts as a filter
>              through which all requests to autoactivate a logi‐ cal
>              volume on this machine are passed. A logical volume is
>              autoactivated if it matches an item in the list. Volumes
>              must also pass the volume_list filter, if present. Tags
>              must be preceded by @ and are checked against all tags
>              defined in the logical volume and volume group metadata
>              for a match. @* is short-hand to check every tag set on
>              the host machine (see tags above).  Logical volume and
>              volume groups can also be included in the list by name
>              e.g. vg00, vg00/lvol1.

Setting `auto_set_activation_skip = 0` in lvm.conf fixes the issue:

    micky@lina:~$ lxc init devuan test
    Creating test done.
    micky@lina:~$ 

    root@lina:~# lvscan | grep test
      ACTIVE            '/dev/lxc/test' [10.00 GiB] inherit
    root@lina:~# 

IMHO it's better to revert LV creation using `lvcreate -kn` as in lxd version <= 0.24 than forcing users to change their LVM settings.

**Comment by stgraber** [comment]
Except that -k is a recent LVM option which doesn't exist in most distributions (specifically Ubuntu 14.04 LTS).

**Comment by mickydelfavero** [comment]
And so using

    func (s *storageLvm) createSnapshotLV(lvname string, origlvname string, readonly bool) (string, error) {
            output, err := s.tryExec(
                    "lvcreate",
                    "-n", lvname,
                    "-s", fmt.Sprintf("/dev/%s/%s", s.vgName, origlvname))
            if err != nil {

it's not an option?

**Comment by stgraber** [comment]
It may be an option, I'm not familiar enough with LVM. @mikemccracken ?

**Comment by alitvak69** [comment]
So what is the plan here.  Can you revert to pre 0.25 lvm options?

**Comment by stgraber** [comment]
No, reverting to -k isn't an option as that'd cause testing regressions and completely break LVM on 14.04. We may be able to go with @mickydelfavero's suggestion if -k and -a aren't actually useful (simply not passing them).

**Comment by mikemccracken** [comment]
`lvcreate -kn` or equivalently, passing `--setactivationskip n` to the lvchange we do after the create is necessary for LVM post-trusty, because the defaults are to skip auto-activation, which breaks our flow.

I think the correct solution here is to know the LVM version and add the flag conditionally.

**Comment by stgraber** [comment]
ok, yeah, making this conditional on the version would be fine, we already have it in the struct too

**Comment by alitvak69** [comment]
I would be happy to test the fix when it is avaialble.

**Comment by alitvak69** [comment]
I pulled new master for lxd, lxc, and lxcfs. The issue still persists.
```
t=2016-01-09T04:02:31+0000 lvl=info msg=handling ip=@ method=POST url=/1.0/containers
t=2016-01-09T04:02:31+0000 lvl=dbug msg="\n\t{\n\t\t\"name\": \"webtest22\",\n\t\t\"profiles\": [\n\t\t\t\"default\"\n\t\t],\n\t\t\"source\": {\n\t\t\t\"fingerprint\": \"e62ce217535634bb8a78238f928f4b57e70b4f4639f049b599d350a278ba136e\",\n\t\t\t\"type\": \"image\"\n\t\t}\n\t}"
t=2016-01-09T04:02:31+0000 lvl=dbug msg="Responding to container create"
t=2016-01-09T04:02:31+0000 lvl=dbug msg="New task operation: 57b411c7-1d08-4fde-9f13-ae40347d3e1b"
t=2016-01-09T04:02:31+0000 lvl=dbug msg="Started task operation: 57b411c7-1d08-4fde-9f13-ae40347d3e1b"
t=2016-01-09T04:02:31+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"async\",\n\t\t\"status\": \"OK\",\n\t\t\"status_code\": 100,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"57b411c7-1d08-4fde-9f13-ae40347d3e1b\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-09T04:02:31.23735626Z\",\n\t\t\t\"updated_at\": \"2016-01-09T04:02:31.23735626Z\",\n\t\t\t\"status\": \"Running\",\n\t\t\t\"status_code\": 103,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest22\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"\"\n\t\t},\n\t\t\"operation\": \"/1.0/operations/57b411c7-1d08-4fde-9f13-ae40347d3e1b\"\n\t}"
t=2016-01-09T04:02:31+0000 lvl=info msg=handling method=GET url=/1.0/operations/57b411c7-1d08-4fde-9f13-ae40347d3e1b/wait ip=@
t=2016-01-09T04:02:31+0000 lvl=dbug msg=ContainerCreateFromImage driver=storage/lvm imageFingerprint=e62ce217535634bb8a78238f928f4b57e70b4f4639f049b599d350a278ba136e name=webtest22 isPrivileged=true
t=2016-01-09T04:02:36+0000 lvl=dbug msg=containerDeleteSnapshots container=webtest22
t=2016-01-09T04:02:36+0000 lvl=dbug msg=ContainerDelete driver=storage/lvm container=webtest22
t=2016-01-09T04:02:41+0000 lvl=dbug msg="Could not remove LV" driver=storage/lvm output="  Failed to find logical volume \"lxc/webtest22\"\n" lvname=webtest22
t=2016-01-09T04:02:41+0000 lvl=dbug msg="Failure for task operation: 57b411c7-1d08-4fde-9f13-ae40347d3e1b: Error mounting snapshot LV: no such file or directory"
t=2016-01-09T04:02:41+0000 lvl=dbug msg="\n\t{\n\t\t\"type\": \"sync\",\n\t\t\"status\": \"Success\",\n\t\t\"status_code\": 200,\n\t\t\"metadata\": {\n\t\t\t\"id\": \"57b411c7-1d08-4fde-9f13-ae40347d3e1b\",\n\t\t\t\"class\": \"task\",\n\t\t\t\"created_at\": \"2016-01-09T04:02:31.23735626Z\",\n\t\t\t\"updated_at\": \"2016-01-09T04:02:31.23735626Z\",\n\t\t\t\"status\": \"Failure\",\n\t\t\t\"status_code\": 400,\n\t\t\t\"resources\": {\n\t\t\t\t\"containers\": [\n\t\t\t\t\t\"/1.0/containers/webtest22\"\n\t\t\t\t]\n\t\t\t},\n\t\t\t\"metadata\": null,\n\t\t\t\"may_cancel\": false,\n\t\t\t\"err\": \"Error mounting snapshot LV: no such file or directory\"\n\t\t},\n\t\t\"operation\": \"\"\n\t}"
```

**Comment by stgraber** [comment]
what's the "lxc info" output on your system?
